core/src/main/java/tc/oc/pgm/command/TimeLimitCommand.java

```diff
@@ -31,26 +31,29 @@ public void timelimit(
       @Nullable Duration overtime,
       @Nullable Duration maxOvertime,
       @Nullable Duration endOvertime) {
+    final TimeLimitMatchModule time = match.getModule(TimeLimitMatchModule.class);
 
-    final TimeLimitMatchModule time = match.needModule(TimeLimitMatchModule.class);
+    if (time != null) {
```

Comment:
The time limit module should _always_ be loaded so this is probably not needed? If its not loaded it should be considered a more fatal error, since it by design always loads even though no time limit is specified in the xml.